examples/multiagent/iql.py

```diff
@@ -0,0 +1,283 @@
+import time
+
+import torch
+
+import wandb
+from tensordict.nn import TensorDictModule
+from torch import nn
+from torchrl.collectors import SyncDataCollector
+from torchrl.data import TensorDictReplayBuffer
+from torchrl.data.replay_buffers.samplers import SamplerWithoutReplacement
+from torchrl.data.replay_buffers.storages import LazyTensorStorage
+from torchrl.envs.libs.vmas import VmasEnv
+from torchrl.envs.utils import ExplorationType, set_exploration_type
+from torchrl.modules import EGreedyWrapper, QValueModule, SafeSequential
+from torchrl.modules.models.multiagent import MultiAgentMLP
+from torchrl.objectives import DQNLoss, SoftUpdate, ValueEstimators
+from torchrl.record.loggers import generate_exp_name
+from torchrl.record.loggers.wandb import WandbLogger
+from utils.logging import log_evaluation, log_training
+
+
+def rendering_callback(env, td):
+    env.frames.append(env.render(mode="rgb_array", agent_index_focus=None))
+
+
+def train(seed):
+    # Device
+    training_device = "cpu" if not torch.has_cuda else "cuda:0"
+    vmas_device = training_device
+
+    # Seeding
+    seed = seed
+    torch.manual_seed(seed)
+
+    # Log
+    log = True
+
+    # Sampling
+    frames_per_batch = 60_000  # Frames sampled each sampling iteration
+    max_steps = 100
+    vmas_envs = frames_per_batch // max_steps
+    n_iters = 500  # Number of sampling/training iterations
+    total_frames = frames_per_batch * n_iters
+    memory_size = frames_per_batch
+
+    scenario_name = "balance"
+    env_config = {
+        "n_agents": 3,
+    }
+
+    config = {
+        # RL
+        "gamma": 0.9,
+        "seed": seed,
+        # Sampling,
+        "frames_per_batch": frames_per_batch,
+        "max_steps": max_steps,
+        "vmas_envs": vmas_envs,
+        "n_iters": n_iters,
+        "total_frames": total_frames,
+        "memory_size": memory_size,
+        "vmas_device": vmas_device,
+        # Training
+        "num_epochs": 45,  # optimization steps per batch of data collected
+        "minibatch_size": 4096,  # size of minibatches used in each epoch
+        "lr": 5e-5,
+        "max_grad_norm": 40.0,
+        "training_device": training_device,
+        # Target
+        "tau": 0.005,
+        # Evaluation
+        "evaluation_interval": 20,
+        "evaluation_episodes": 200,
+    }
+
+    model_config = {
+        "shared_parameters": True,
+    }
```

Comment:
Would be nice to have a hydra config
```
collector:
  frames_per_batch: 60_000
  max_steps: 100
  ...
objective:
  gamma: 0.9
  ...
```

Reply:
Sure, but is it possible to do that in the same file?
I would really like to have every file be self contained (the config too)